### DISCUSSION TITLE
Is there an end-of-life of shapely-1?

### DISCUSSION THREAD

**Discussion by ajolma** [shapely/shapely#1984] [General]
Is there an end-of-life for shapely version 1? Or, maybe, when would it be reasonable for a big project to switch to shapely 2? The project I'm working on is a bit formal one so if I suggest a switch I think I need a some more or less official statement.

**Comment by jorisvandenbossche** [comment]
I would say it's more than time to switch to shapely 2.0. In general the modifications you would need to do to make your code run with 2.0 should only be minor, and it should also be easily possible to make your code work with both shapely 1.8 and 2.0, if that is desirable. 
See the [release notes](https://shapely.readthedocs.io/en/stable/release/2.x.html#shapely-2-0-api-changes-deprecated-in-1-8) (with more details at https://shapely.readthedocs.io/en/stable/migration.html) and the [section on breaking changes in 2.0](https://shapely.readthedocs.io/en/stable/release/2.x.html#breaking-api-changes)

We haven't discussed officially marking shapely 1.x as "end of life", but in practice we are (at the moment) not planning to do any maintenance on that branch (we also haven't made any release for 1.8.x since 2022, and for example there are no binary wheels for that release for Python 3.12 as a result)
